### PR TITLE
added oauth

### PR DESCRIPTION
**Pull request by Skaelv**
- New Design (by Tonia)
- Oauth workflow to get app_id and secret
- Enable secure mode checkbox

### CODE REVIEW

File: bootstrap.php [hunk 1]
```diff
@@ -78,51 +78,106 @@ public function dismissibleMessage($text)
...
-
-    if ($_GET['appId']) {
-      $app_id = WordPressEscaper::escAttr($_GET['appId']);
-      $dismissable_message = $this->dismissibleMessage('We’ve copied your new Intercom app ID below. Click to save changes and then close this window to finish signing up for Intercom.');
+    if ($_GET['enable_secure_mode']) {
+      $dismissable_message = $this->dismissibleMessage('Secure Mode successfully enabled');
     }
 
     return <<<END
-<div class="wrap">
```
Comment (bobjflong) [inline bootstrap.php]: Can we double check that no unescaped content is going into the HTML? PHP won't automatically escape like rails :)

File: bootstrap.php [hunk 2]
```diff
@@ -130,9 +185,39 @@ public function htmlClosed()
...
+      } else if (window.oauth_success === undefined) {
+        setTimeout(checkWindow, 200);
+      }
+    };
+    checkWindow();
+  });
+  $('#intercom-secure-mode').unbind('click').click(function() {
+    $('#intercom-secure-mode').prop('checked', false);
+    if(confirm('Are you sure you want to enable secure mode for Intercom ?'))  {
+      window.location = '/wp-admin/options-general.php?page=intercom&enable_secure_mode=1';
```
Comment (bobjflong) [inline bootstrap.php]: I think this should be a POST request that checks the the nonce as normal.

Comment (Skaelv) [inline bootstrap.php]: I'm actually not using this anymore I'll remove it. The auth is done in : https://github.com/intercom/wordpress-oauth/pull/1/files

File: bootstrap.php (hunk 1 above)
Comment (Skaelv) [inline bootstrap.php]: Is it something that was done and I broke or should I implement it ?

Comment (bobjflong) [inline bootstrap.php]: Yeah I just mean to check that any `$variables` used in the html are not user submitted

Comment (Skaelv) [inline bootstrap.php]: Ok I get it :) I'll double check

**Comment by Skaelv** [comment]
- Double checked the html for unescaped content.
- added _wpnonce check for secure mode post
Ready to merge ?

**Comment by RuairiK** [comment]
:ship:

**Pull request merged**